drivers/include/mtd.h

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (C) 2016  OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_mtd Memory Technology Device
+ * @ingroup     drivers
+ * @brief       Low level Memory Technology Device interface
+ *
+ * Generic memory technology device interface
+ *
+ * @{
+ *
+ * @file
+ *
+ * @author      Aurelien Gonce <aurelien.gonce@altran.com>
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ */
+
+#ifndef MTD_H
+#define MTD_H
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name Disk Status Bits
+ * @{
+ */
+typedef enum {
+    MTD_STA_NOINIT  = 0x00, /**< Drive not initialized */
+    MTD_STA_INIT    = 0x01, /**< Drive initialized */
+    MTD_STA_NODISK  = 0x02, /**< No medium in the drive */
+    MTD_STA_PROTECT = 0x04  /**< Write protected */
+} mtd_sta_t;
+/** @} */
+
+/**
+ * @name Command code for disk_ioctrl function
+ * @{
+ */
+
+/**
+ * @name Generic ioctl command
+ * @{
+ */
+#define MTD_CTRL_SYNC           0   /**< Flush disk cache (for write functions) */
+#define MTD_CTRL_ERASE_BLOCK    6   /**< Force erased a block of blocks */
```

Comment:
Read block of sectors. But I'm not convinced this is needed. A more intelligent erase function should be able to use block erase or chip erase if needed.